Лаб. 3 6411 Шехматов Виктор

Nakamo312 opened the pull request:
ФИО: Шехматов Виктор Александрович
Номер группы: 6411
Номер лабораторной: 3
Номер варианта: 25
Краткое описание предметной области: Поликлиника.
Краткое описание добавленных фич: В проект была интегрирована поддержка PostgreSQL через Entity Framework Core. Реализован AppDbContext, содержащий DbSet для всех доменных сущностей (Patient, Doctor, Specialization, Appointment).

Настроена система миграций EF Core для автоматического создания и обновления структуры базы данных при запуске приложения. 
Добавлен механизм начального заполнения данных через класс EfDataSeeder, который выполняет первичную инициализацию базы тестовыми данными из DataSeed.

В инфраструктуру включен проект Clinic.ServiceDefaults, обеспечивающий:
    Интеграцию с OpenTelemetry для сбора телеметрии
    Health Checks для мониторинга состояния сервисов
    Service Discovery для автоматического обнаружения сервисов

Оркестрация всех компонентов приложения реализована через платформу .NET Aspire, которая объединяет API, базу данных и системы мониторинга в единую среду выполнения с автоматическим управлением зависимостями и конфигурацией

danlla commented on Pharmacies/Pharmacies.Application/Services/PositionService.cs:
```diff
@@ -0,0 +1,92 @@
+public class PositionService(
+    IRepository<Position, int> positionRepository,
+    IRepository<ProductGroup, int> productGroupRepository,
+    IRepository<PharmaceuticalGroup, int> pharmaceuticalGroupRepository,
```
Параметр не используется

danlla commented on Pharmacies/Pharmacies.Application/Services/PositionService.cs:
```diff
@@ -0,0 +1,92 @@
+
+    public async Task Update(int key, PositionDto entityDto)
+    {
+        var position = await positionRepository.GetByKey(key);
+        if (position == null)
+        {
+            throw new Exception($"Position with key {key} not found.");
+        }
```
```csharp
var position = await positionRepository.GetByKey(key) ?? throw new Exception($"Position with key {key} not found.");
```

danlla commented on Pharmacies/Pharmacies.Application/Services/PriceService.cs:
```diff
@@ -0,0 +1,58 @@
+
+    public async Task Update(int key, PriceDto entityDto)
+    {
+        var price = await priceRepository.GetByKey(key);
+        if (price == null)
+        {
+            throw new Exception($"Price with key {key} not found.");
+        }
```
```csharp
var price = await priceRepository.GetByKey(key) ?? throw new Exception($"Price with key {key} not found.");
```

danlla commented on Pharmacies/Pharmacies.Application/Services/PriceService.cs:
```diff
@@ -0,0 +1,58 @@
+
+    public async Task Delete(int key)
+    {
+        var price = await priceRepository.GetByKey(key);
+        if (price == null)
+        {
+            throw new Exception($"Price with key {key} not found.");
+        }
```
```csharp
if (await priceRepository.GetByKey(key) == null)
    throw new Exception($"Price with key {key} not found.");
```

danlla commented:
А еще стоит поменять `ФИО` в описании пр)

Nakamo312 committed:
initial commit

Nakamo312 committed:
fix test.yml

Nakamo312 committed:
add comments for IModel.cs

Nakamo312 committed:
change field IsRepeated in Appointment data model

Nakamo312 committed:
added Repository class. Added DataBase context class

Nakamo312 committed:
create Application Layer: -Added MappingProfile;\n-Added DTOs;

Nakamo312 committed:
created Base Controller and Services

Nakamo312 committed:
modified DTOs

Nakamo312 committed:
Add API Controllers

Nakamo312 committed:
Add InMmempry Repositories

Nakamo312 committed:
Added API CRUD endpoints, added DTO representation schemes, addrd logging through middleware

Nakamo312 committed:
update README.md

Nakamo312 committed:
changes status code and response header in Create methods

Nakamo312 committed:
Added analytic services and controllers

Nakamo312 committed:
rm file

Nakamo312 committed:
fix test.yml

Nakamo312 committed:
fix test.yml

Nakamo312 committed:
remove exceptions messages from endpoints

Nakamo312 committed:
change swager gen documentation for DTO schemes. Change content-type for all endpoints

Nakamo312 committed:
refactor: decouple API layer from domain models

- Remove Domain.Model dependency from base controller
- Use reflection for ID access in DTOs instead of Model inheritance
- Keep API layer dependent only on DTOs and application contracts

Nakamo312 committed:
Remarks have been fixed

Nakamo312 committed:
Remarks have been fixed

Nakamo312 committed:
Remarks have been fixed

Nakamo312 committed:
Remarks have been fixed

Nakamo312 committed:
fix regex in PatientUpdateDto

Nakamo312 committed:
Improve error management: hide detailed validation errors

Nakamo312 committed:
revert last changes

Nakamo312 committed:
change response stats code in DELETE methods

Nakamo312 committed:
Aspire mood orchestrator; creating a conference for the database

Nakamo312 committed:
update README.MD

Nakamo312 committed:
comments corrected

Nakamo312 committed:
rename folder

Nakamo312 committed:
rename folder

Nakamo312 committed:
rename folder

Nakamo312 committed:
rename folder

Nakamo312 committed:
rename folder

Nakamo312 committed:
rename folder

Nakamo312 committed:
registred DataSeed in DI container

Nakamo312 committed:
update phone number regex

Nakamo312 committed:
added phone number normalizatiom; change contracts for PUT methods

Nakamo312 committed:
changes namespaces

Nakamo312 committed:
changes namespaces

Nakamo312 committed:
modified README.md

Nakamo312 committed:
change Domain Models type field from uint to Guid

Nakamo312 committed:
refactor Program.cs

Nakamo312 reviewed (commented):
Переделал на Guid, по той причине что при вставке данных после сидирования возникли ошибки с тем что Id икрементируется  начиная с нуля. Можно было конечно создать миграцию на заполнение БД, но добавлять для этого HasData() каждой сущности в AppDbContext очень сильно кажется кривым, и тот факт что данные  должны быть определены статически.

Вместо этого решил заполнение делать через сидер после применения миграций в рантайме.  Тем более будет проще потом сделать 4 лабораторную не меняя архитектуру. А вобще хотелось бы услышать совет как это все таки правильно делается

danlla commented on Clinic/Clinic.Infrastructure/Repositories/Repository.cs:
```diff
@@ -0,0 +1,73 @@
+/// <typeparam name="T">The type of entity this repository works with, must inherit from Model.</typeparam>
+public class Repository<T>(AppDbContext context) : IRepository<T> where T : Model
+{
+    private readonly AppDbContext _context = context ?? throw new ArgumentNullException(nameof(context));
```
`context` не null по объявлению
тут можно избавиться от поля `_context` и использовать `context`

danlla reviewed (changes requested):
В postgres по-хорошему чтобы таблицы и столбы наименовались в `snake_case`, нужно это поправить

Nakamo312 committed:
Rename tablenames and columns
